src/components/Modals/FiatRamps/fiatRampProviders/gem.ts

```diff
@@ -0,0 +1,127 @@
+import { adapters } from '@shapeshiftoss/caip'
+import axios from 'axios'
+import { getConfig } from 'config'
+import flatten from 'lodash/flatten'
+import memoize from 'lodash/memoize'
+import uniqBy from 'lodash/uniqBy'
+import queryString from 'querystring'
+
+import { FiatRampAction, FiatRampCurrency } from '../FiatRampsCommon'
+
+enum TransactionDirection {
+  BankToBlockchain = 'bank_blockchain',
+  CardToBlockchain = 'card_blockchain',
+  BlockchainToBank = 'blockchain_bank',
+}
+
+// Non-exhaustive typings. We do not want to keep this a 1/1 mapping to an external API
+// There could be breaking changes with other fields and that's fine, these are the only ones we need
+export type GemCurrency = {
+  gem_asset_id: string
+  name: string
+  ticker: string
+  assetId: string
+  disabled?: boolean
+}
+
+export type SupportedCurrency = {
+  destination: {
+    currencies: GemCurrency[]
+  }
+  source: {
+    currencies: GemCurrency[]
+  }
+  transaction_direction: TransactionDirection
+}
+
+export const fetchCoinifySupportedCurrencies = memoize(async (): Promise<SupportedCurrency[]> => {
+  try {
+    const { data } = await axios.get(getConfig().REACT_APP_GEM_COINIFY_SUPPORTED_COINS)
+    return data
+  } catch (e: any) {
+    console.error(e)
+    return []
+  }
+})
+
+export const fetchWyreSupportedCurrencies = memoize(async (): Promise<SupportedCurrency[]> => {
+  try {
+    const { data } = await axios.get(getConfig().REACT_APP_GEM_WYRE_SUPPORTED_COINS)
+    return data
+  } catch (e: any) {
+    console.error(e)
+    return []
+  }
+})
+
+export const isBuyAsset = (currency: SupportedCurrency) =>
+  currency.transaction_direction === TransactionDirection.BankToBlockchain ||
+  currency.transaction_direction === TransactionDirection.CardToBlockchain
+
+export const isSellAsset = (currency: SupportedCurrency) =>
+  currency.transaction_direction === TransactionDirection.BlockchainToBank
+
+export const parseGemSellAssets = memoize((assets: SupportedCurrency[]): FiatRampCurrency[] =>
+  parseGemAssets(
```

Comment:
Oh damn, we don't have the `no-use-before-define` rule on eslint, it makes it a bit harder to understand as you read the use before the definition

(Just wanted to leave it there)